### PR TITLE
don't use build tags to mark integration tests

### PR DESCRIPTION
**Pull request by mikedanese**
This is a bad pattern that leads to checked in code that isn't check for compilation. We should avoid this if it doesn't provide value, which it seems like it doesn't.

### CODE REVIEW

**Comment by mikedanese** [comment]
Aaaand this is why we shouldn't use build tags for this stuff.

**Comment by ixdy** [comment]
What's the status of this PR? I'm not sure if I should be reviewing this or not, given that there are build failures.

**Comment by ixdy** [comment]
we still interested in this one too?

**Comment by ixdy** [comment]
yeah. @mikedanese @roberthbailey @zmerlynn @eparis can you help me out?

**Comment by ixdy** [comment]
^ oh that's where this comment went. sorry, wrong PR.

**Comment by spxtr** [comment]
Please rebase this.

**Comment by spxtr** [comment]
[This](https://github.com/kubernetes/kubernetes/blob/274df99e9bf63213325d9813eec84288d3c5ab73/hack/make-rules/test-integration.sh#L71) and [this](https://github.com/kubernetes/kubernetes/blob/274df99e9bf63213325d9813eec84288d3c5ab73/test/integration/doc.go#L17-L19) will need updating. Right now we have `no-etcd` and `no-docker` tags that may need their behavior observed.

**Comment by spxtr** [comment]
Please fix golint.

**Comment by mikedanese** [comment]
@spxtr done.

**Comment by spxtr** [comment]
```
I0427 08:45:19.794] FAILED   hack/make-rules/../../hack/verify-golint.sh	37s
I0427 08:45:19.794] Skipping hack/make-rules/../../hack/verify-govet.sh
I0427 08:45:19.794] Verifying hack/make-rules/../../hack/verify-import-boss.sh
W0427 08:45:19.894] Some packages passed golint but are not listed in hack/.linted_packages.
W0427 08:45:19.895] Please add them in alphabetical order:
W0427 08:45:19.895] 
W0427 08:45:19.895]   echo test/integration/client >> hack/.linted_packages
W0427 08:45:19.895]   echo test/integration/configmap >> hack/.linted_packages
W0427 08:45:19.895]   echo test/integration/defaulttolerationseconds >> hack/.linted_packages
W0427 08:45:19.895]   echo test/integration/etcd >> hack/.linted_packages
W0427 08:45:19.895]   echo test/integration/garbagecollector >> hack/.linted_packages
W0427 08:45:19.895]   echo test/integration/kubectl >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/pods >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/quota >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/secrets >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/serviceaccount >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/storageclasses >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/thirdparty >> hack/.linted_packages
W0427 08:45:19.896]   echo test/integration/ttlcontroller >> hack/.linted_packages
W0427 08:45:19.896]   LC_ALL=C sort -o hack/.linted_packages hack/.linted_packages
W0427 08:45:19.896] 
W0427 08:45:19.896] You can test via this script and commit the result.
```

**Comment by spxtr** [comment]
This PR makes a strong case that these build tags were a mistake:

1. Lots of completely 100% dead code that isn't even close to compiling.
1. Lots of code that fails golint.
1. Lots of code that fails go vet.

File: hack/benchmark-integration.sh [hunk 1]
```diff
@@ -33,14 +33,14 @@ ARGS="-bench-pods 3000 -bench-tasks 100 -bench-tasks 10"
 runTests() {
   kube::etcd::start
   kube::log::status "Running benchmarks"
-  KUBE_GOFLAGS="-tags 'benchmark no-docker' -bench . -benchmem -benchtime 1s -cpu 4" \
+  KUBE_GOFLAGS="-tags -bench . -benchmem -benchtime 1s -cpu 4" \
```
Comment (spxtr) [inline hack/benchmark-integration.sh]: -tags?

**Comment by ixdy** [comment]
yeah, this is a mess:
```
I0428 20:28:00.788] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:106: undefined: framework.StartPods
I0428 20:28:00.789] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:113: cannot use false (type bool) as type float32 in field value
I0428 20:28:00.789] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:113: too many values in struct initializer
I0428 20:28:00.790] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:116: undefined: s
I0428 20:28:00.790] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:116: undefined: t
I0428 20:28:00.791] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:117: undefined: s
I0428 20:28:00.791] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:117: undefined: t
I0428 20:28:00.792] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:127: m.RestClient undefined (type *framework.MasterComponents has no field or method RestClient)
I0428 20:28:00.792] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:140: cannot use labels.Everything() (type labels.Selector) as type string in field value
I0428 20:28:00.793] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:141: cannot use fields.OneTermEqualSelector("k8s.io/kubernetes/pkg/api".PodHostField, host) (type fields.Selector) as type string in field value
I0428 20:28:00.793] k8s.io/kubernetes/test/integration/master/master_benchmark_test.go:141: too many errors
```

**Comment by mikedanese** [comment]
@k8s-bot bazel test this

**Pull request merged**